src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticService.cs

```diff
@@ -0,0 +1,118 @@
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeLocalFunctionStatic
+{
+    [Shared]
+    [ExportLanguageService(typeof(MakeLocalFunctionStaticService), LanguageNames.CSharp)]
+    internal sealed class MakeLocalFunctionStaticService : ILanguageService
+    {
+        private static readonly char[] s_underscore = { '_' };
+        private readonly SyntaxGenerator s_generator = CSharpSyntaxGenerator.Instance;
+
+        internal async Task<Document> CreateParameterSymbolAsync(Document document, LocalFunctionStatementSyntax localfunction, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(true);
+            var localFunctionSymbol = semanticModel.GetDeclaredSymbol(localfunction, cancellationToken);
+            var dataFlow = semanticModel.AnalyzeDataFlow(localfunction);
+            var captures = dataFlow.CapturedInside;
+
+            var parameters = CreateParameterSymbol(captures);
+
+            //Finds all the call sites of the local function
+            var workspace = document.Project.Solution.Workspace;
+            var arrayNode = await SymbolFinder.FindReferencesAsync(localFunctionSymbol, document.Project.Solution, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            var rootOne = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var editor = new SyntaxEditor(rootOne, s_generator);
+
+
+            foreach (var referenceSymbol in arrayNode)
+            {
+                foreach (var location in referenceSymbol.Locations)
+                {
+                    var root = await location.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                    var syntaxNode = root.FindNode(location.Location.SourceSpan); //Node for the identifier syntax
+
+
+                    var invocation = (syntaxNode as IdentifierNameSyntax).Parent as InvocationExpressionSyntax;
+                    if (invocation == null)
+                    {
+                        return document;
+                    }
+
+                    var argList = invocation.ArgumentList;
+                    List<ArgumentSyntax> x = new List<ArgumentSyntax>();
+
+                    foreach (var parameter in parameters)
+                    {
+
+                        var newArgument = GenerateArgument(parameter, parameter.Name, false);
```

Comment:
Why `shouldUseNamedArguments` is always `false`? What if the invocation contains named argument already?

Reply:
It would only be true if we were passing in constants, we decided to take it out since it was an optional variable. We are passing the local variables explicitly

Reply:
What I meant is, if named argument is specified out of order, then added argument must be named as well, otherwise it's a compiler error.

For example, if you added another parameter `int c` to the local function, then the invocation needs to be `Foo(b: 1, a: 2, c: 3)`
```cs
        public void Baz()
        {
            Foo(b: 1, a: 2);

            void Foo(int a, int b)
            { }
        }
```